### PR TITLE
chore(uplink) fix unit tests

### PR DESCRIPTION
**Pull request by dduportal**
This PR is a fixup of https://github.com/jenkins-infra/helm-charts/pull/686

cc @Sathwik-git for information (as part of the learning):

- Unit tests were not in the proper location expected by the CI system. I should never had approved #686 
- Asserts with matchregexp are to be avoided at all costs unless there is no other solution. I've changed to `equal` in the uplink unittests
- @Sathwik-git you never ran the command `helm unittest .` or it never showed any test run: the custom values required a proper syntax for the YAML path (using a `['key']` syntax)

### CODE REVIEW

File: charts/uplink/tests/defaults_test.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - secret.yaml
+tests:
+  - it: should render with default values without error
+    asserts:
+      - isKind:
+          of: Secret
+      - equal:
+          path: metadata.name
+          value: RELEASE-NAME-uplink
+      - notExists:
```
Comment (smerle33) [inline charts/uplink/tests/defaults_test.yaml:12]: I got an error running the test locally from this PR branch :

```
helm unittest --helm3 uplink       
 FAIL   uplink/tests/defaults_test.yaml
        - Execution Error: 
                Assertion type `notExists` is invalid


### Error:  Assertion type `notExists` is invalid


Charts:      1 failed, 1 errored, 0 passed, 1 total
Test Suites: 1 failed, 1 errored, 0 passed, 1 total
Tests:       0 passed, 0 total
Snapshot:    0 passed, 0 total
Time:        1.064083ms

Error: plugin "unittest" exited with error
```

Comment (dduportal) [reply charts/uplink/tests/defaults_test.yaml:12]: Can you check you have the latest 0.3.4 unittest plugin installed?

```
helm plugin ls | grep unit
unittest        0.3.4           Unit test for helm chart in YAML with ease to keep your chart functional and robust.
```

Comment (smerle33) [reply charts/uplink/tests/defaults_test.yaml:12]:
```
helm plugin ls | grep unit
unittest        0.2.8   Unit test for helm chart in YAML with ease to keep your chart functional and robust.
```

Comment (smerle33) [reply charts/uplink/tests/defaults_test.yaml:12]: after a `helm plugin update unittest`
I got 
```
helm unittest uplink        

### Chart [ uplink ] uplink

 PASS  test secret.yaml uplink/tests/custom_values_test.yaml
 PASS  test default behavior of secret.yaml with default values uplink/tests/defaults_test.yaml

Charts:      1 passed, 1 total
Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshot:    0 passed, 0 total
Time:        3.526209ms
```

**Review by smerle33 — APPROVED** [review]
🚀

File: charts/uplink/tests/defaults_test.yaml (hunk 1 above)
Comment (dduportal) [reply charts/uplink/tests/defaults_test.yaml:12]: you need to update. Be careful, as this plugins changed their GitHub repository (it was a fork from quintus, merged back to helm unit organization).

Start with 

```
helm plugin uninstall unittest
```

Then follow instruction at https://github.com/helm-unittest/helm-unittest to get the 0.3.5

Comment (dduportal) [reply charts/uplink/tests/defaults_test.yaml:12]: > after a `helm plugin update unittest` I got
> 
> ```
> helm unittest uplink        
> 
> ### Chart [ uplink ] uplink
> 
>  PASS  test secret.yaml uplink/tests/custom_values_test.yaml
>  PASS  test default behavior of secret.yaml with default values uplink/tests/defaults_test.yaml
> 
> Charts:      1 passed, 1 total
> Test Suites: 2 passed, 2 total
> Tests:       2 passed, 2 total
> Snapshot:    0 passed, 0 total
> Time:        3.526209ms
> ```

Nice job!

**Pull request merged**